compiler/rustc_middle/src/ty/consts/int.rs

```diff
@@ -398,6 +398,11 @@ impl ScalarInt {
     pub fn to_f128(self) -> Quad {
         self.to_float()
     }
+
+    #[inline]
+    pub fn bitwise_not(self) -> Self {
+        Self { data: self.size().truncate(!self.data), size: self.size }
```

Comment:
Oh, are bools represented as an i8 scalar int rather than an i1?

Reply:
I think so, for reference: https://github.com/rust-lang/rust/blob/3002af6cb643138839537f6fd0265162610fdbbe/compiler/rustc_middle/src/ty/consts/int.rs#L170-L171 and https://github.com/rust-lang/rust/blob/3002af6cb643138839537f6fd0265162610fdbbe/compiler/rustc_middle/src/ty/consts/int.rs#L325-L331.

Reply:
Can you add some comments telling us not use it when the type is bool? Or add a type parameter?

Reply:
> > IIUC, we need additional handling for bool. For the bitwise-not of a bool, `0x00` should be `0x01` instead of `0xff`.
> 
> Is this related to the i1 representation we use for bool immediates?

I'm not sure, but I think the i1 representation is only at the codegen phase.